torch/optim/adam.py

```diff
@@ -416,14 +415,14 @@ def _single_tensor_adam(params: List[Tensor],
 
             param.addcdiv_(exp_avg, denom)
         else:
-            step = step_t.item()
+            step = step_t
 
             bias_correction1 = 1 - beta1 ** step
```

Comment:
Converting step to a tensor induces the last remaining test failure, any idea why that might be @albanD ? I played around with `torch.pow` and it looks like there is a difference between scalar exponentiation and torch exponentiation that differs from the fused Adam impl. (Maybe fused Adam matched the scalar pow?)

Reply:
scalar exponentiation is done in `double` precision, whereas default tensorization is float32. exponentiation is particularly sensitive to the reduced precision. I'd verify this by moving `step` to `float64`

Reply:
Thanks! I will try this

Reply:
Hmm this didn't work, the accuracy actually went down (I got NaNs). Originally 22/64 elements are mismatches, and now they all are.

Reply:
Ignore my previous comment, I confirmed this by casting to double at the exponentiation (I just cast step at line 422) and it worked. Should I try and make step float64 by default, or increase the tolerance of the test